test/functional/test_framework/test_framework.py

```diff
@@ -203,67 +197,60 @@ def main(self):
 
     # Public helper methods. These can be accessed by the subclass test scripts.
 
-    def start_node(self, i, dirname, extra_args=None, rpchost=None, timewait=None, binary=None, stderr=None):
+    def start_node(self, i, dirname, extra_args=[], rpchost=None, timewait=None, binary=None, stderr=None):
```

Comment:
wow I did not know.

In general I think that the user should be able to pass `null` (Nothing), when he means "default value".

This code can be easily fixed by keeping the default value to Nothing and first line, checking if extra_args is nothing, if that is the case, init to an empty array.